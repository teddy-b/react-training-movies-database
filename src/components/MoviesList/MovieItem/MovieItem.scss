// $portrait-phone-max-width: 575px;
$landscape-phone-min-width: 576px;
// $landscape-phone-max-width: 767px;
// $tablet-min-width: 768px;
// $tablet-max-width: 991px;
$small-desktop-min-width: 992px;
// $small-desktop-max-width: 1199px;
$large-desktop-min-width: 1200px;

.movieItem {
  width: 100vw;
  margin: 0;
}

.poster {
  width: 100%;
}

.movieTitle {
  text-transform: uppercase;
  font-weight: bold;
}

.movieYear {
  border: 1px solid grey;
  float: right;
}

.movieGenre {
  color: grey;
  flex-grow: 2;
}

@media screen and (min-width: $landscape-phone-min-width) {
  .movieItem {
    width: 44vw;
    margin: 2vw;
  }
}

@media screen and (min-width: $small-desktop-min-width) {
  .movieItem {
    width: 28vw;
  }
}

@media screen and (min-width: $large-desktop-min-width) {
  .movieItem {
    width: 20vw;
  }
}
