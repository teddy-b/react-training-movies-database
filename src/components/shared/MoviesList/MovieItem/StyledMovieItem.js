import styled from 'styled-components'

import {
  grey3,
  landscapePhoneMinWidth,
  smallDesktopMinWidth,
  largeDesktopMinWidth
} from '../../../../constants/styles'

export const StyledMovieItem = styled.figure`
  width: 100vw;
  margin: 0;

  @media screen and (min-width: ${landscapePhoneMinWidth}) {
    width: 44vw;
    margin: 2vw;
  }

  @media screen and (min-width: ${smallDesktopMinWidth}) {
    width: 28vw;
  }

  @media screen and (min-width: ${largeDesktopMinWidth}) {
    width: 20vw;
  }
`

export const StyledImg = styled.img`
  width: 100%;
`

export const StyledMovieYear = styled.div`
  padding: 2px;
  float: right;
  border: 1px solid ${grey3};
`

export const StyledMovieTitle = styled.div`
  text-transform: uppercase;
  font-weight: bold;
`

export const StyledMovieGenre = styled.div`
  color: ${grey3};
  flex-grow: 2;
`
