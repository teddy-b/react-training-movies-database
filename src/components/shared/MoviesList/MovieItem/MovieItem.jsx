/* @flow */

import * as React from 'react'
import { Link } from 'react-router-dom'

import './MovieItem.scss'

type Props = {
  genres: Array<string>,
  id: number,
  onSelectMovie: (id: number) => void,
  poster_path: ?string,
  release_date: string,
  title: string
}

const MovieItem = ({
  genres,
  id,
  onSelectMovie,
  poster_path: posterPath,
  release_date: releaseDate,
  title
}: Props) => (
  <Link
    to={`/film/${id}`}
    // style={{ textDecoration: 'none' }}
  >
    <figure className="movieItem" onClick={() => onSelectMovie(id)}>
      {posterPath && <img className="poster" src={posterPath} alt={title} />}
      <figcaption className="movieInfo">
        <div className="movieYear">{new Date(releaseDate).getFullYear()}</div>
        <div className="movieTitle">{title}</div>
        <div className="movieGenre">{genres.join(', ')}</div>
      </figcaption>
    </figure>
  </Link>
)

export default MovieItem
