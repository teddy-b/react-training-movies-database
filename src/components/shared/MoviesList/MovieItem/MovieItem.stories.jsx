/* @flow */

import * as React from 'react'
import { MemoryRouter } from 'react-router-dom'

import { storiesOf } from '@storybook/react'
import { linkTo } from '@storybook/addon-links'

import type { Renderable } from '@storybook/react'

import MovieItem from './MovieItem'
import moviesMock from '../../../../mocks/movies-mocks'

const singleMovieMock = moviesMock.data[0]
const props = {
  ...singleMovieMock,
  key: singleMovieMock.id,
  onSelectMovie: linkTo('Components/Movie')
}

storiesOf('Components/MovieItem', module)
  .add('default', (): Renderable => (
    <MemoryRouter>
      <MovieItem {...props} />
    </MemoryRouter>
  ))
