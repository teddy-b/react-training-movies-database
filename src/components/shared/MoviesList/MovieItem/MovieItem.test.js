import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import MovieItem from './MovieItem'
import { moviesMock } from '../../../../mocks/movies-mocks'

describe('Movie', () => {
  const singleMovieMock = moviesMock[0]

  it('renders correctly', () => {
    const component = renderer.create(
      <MovieItem
        key={singleMovieMock.id}
        {...singleMovieMock}
        onSelectMovie={jest.fn()}
      />
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should trigger onSelectMovie onClick', () => {
    const onSelectMovieMock = jest.fn()
    const component = shallow(
      <MovieItem
        key={singleMovieMock.id}
        {...singleMovieMock}
        onSelectMovie={onSelectMovieMock}
      />
    )

    component.find('figure').at(0).simulate('click')

    expect(onSelectMovieMock).toHaveBeenCalled()
  })
})
