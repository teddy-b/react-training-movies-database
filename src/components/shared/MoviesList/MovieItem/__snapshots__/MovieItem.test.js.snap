// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie renders correctly 1`] = `
<figure
  className="movieItem"
  onClick={[Function]}
>
  <img
    alt="Fifty Shades Freed"
    className="poster"
    src="https: //image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
  />
  <figcaption
    className="movieInfo"
  >
    <div
      className="movieYear"
    >
      2018
    </div>
    <div
      className="movieTitle"
    >
      Fifty Shades Freed
    </div>
    <div
      className="movieGenre"
    >
      Drama, Romance
    </div>
  </figcaption>
</figure>
`;
